tests/test_pydantic_factory.py

```diff
@@ -14,3 +18,23 @@ class AFactory(ModelFactory[A]):
 
     for _ in range(5):
         assert AFactory.build()
+
+
+@pytest.mark.skipif(sys.version_info <= (3, 9), reason="python 10+ features")
+def test_list_unions() -> None:
+    # issue: https://github.com/litestar-org/polyfactory/issues/300, no error reproduced
+    class A(BaseModel):
+        a: str
+
+    class B(BaseModel):
+        b: str
+
+    class C(BaseModel):
+        c: list[A] | list[B]
+
+    class CFactory(ModelFactory[C]):
+        __model__ = C
+
+    assert isinstance(CFactory.build().c, list)
+    assert len(CFactory.build().c) > 0
+    assert isinstance(CFactory.build().c[0], (A, B))
```

Comment:
would it be okay to build once?
```suggestion
    c_instance = CFactory.build()
    
    assert isinstance(c_instance.c, list), "c is not a list"
    assert len(c_instance.c) > 0, "c list is empty"
    assert isinstance(c_instance.c[0], (A, B)), "c[0] is not an instance of A or B"
```

Reply:
it should, the issue was getting a descriptor.